ts-github-profile-finder/src/types/index.ts

```diff
@@ -0,0 +1,17 @@
+export type UserInfoStatus = "idle" | "pending" | "resolved" | "rejected";
+
+export interface UserInfo {
+  data: UserInfoData | null;
+  status: UserInfoStatus;
+}
```

Comment:
data에 null을 허용하지 않으면서 status: resolved 상태에서만 data가 있음을 보장하려면 이런 방법도 있어!
이렇게 하면 userInfo: UserInfo 에 `if (userInfo.status === 'resolved') {` 절만 있어도 이 안에서 null체크 없이 data를 안전하게 사용할 수 있어.

```suggestion
export type UserInfo = { status: "idle" } | { status: "pending" } | { status: "resolved", data: UserInfoData } | { status: "rejected" };
```

Reply:
와대박.......싱기

Reply:
거녕짱!~

Reply:
오와...